ark/schema/shared/errors.ts

```diff
@@ -279,6 +287,24 @@ export class ArkErrors
 	}
 }
 
+export class TraversalError extends Error {
+	name = "TraversalError"
+	arkErrors!: ArkErrors
+
+	constructor(errors: ArkErrors) {
+		if (errors.length === 1) super(errors.summary)
+		else super("\n" + errors.map(error => `  • ${indent(error)}`).join("\n"))
+
+		Object.defineProperty(this, "arkErrors", {
+			value: errors,
+			enumerable: false
+		})
```

Comment:
The reason for the usage of `defineProperty` is because Node will log extra properties on an `Error`:
```ts
class CustomError extends Error {
    constructor() {
        super("Custom message")
        this.prop = 1;
    }
}

throw new CustomError();
```

it prints `Uncaught [CustomError: Custom message] { prop: 1 }` for me. This can be avoided by making the properties non-enumerable.

I had to name it `arkErrors` instead of `errors` though because it appears that `errors` is included even if non-enumerable. Gah!

Reply:
wtf 🫠